### ISSUE TITLE
Attribute for externally tagged enums or match serde by default

### ISSUE THREAD

**Issue by Heliozoa** [Aleph-Alpha/ts-rs#38]
Hi,
currently the the TS::decl() generated for a type like 
```rs
#[derive(serde::Serialize, TS)]
enum Enum {
    A(&'static str),
    B { s: &'static str },
    C
}
```
looks like this
```ts
type Enum = string | {
    s: string,
} | "C";
```
whereas serde_json generates output matching the type
```ts
type Enum = { "A": string } | { "B": { s: string } } | "C"
```
`serde-compat` allows for matching other styles of representing enums, like `#[serde(tag = "tag", content = "content")]`, but there's no way to match the default since there's no serde attribute for it. I think matching serde's default output is preferable since it allows for differentiating between `A("C")` and `C`, but an attribute would work too.

**Comment by NyxCode** [comment]
I'm currently working on this.
In case you want to help, a few tests testing this behaviour, including its interaction with #[ts(inline)], would be tremendously usefull!

**Comment by NyxCode** [comment]
What's working so far:
```rust
#[derive(TS)]
enum SimpleEnum {
    A(String),
    B(i32),
    C,
    D(String, i32),
    E(Foo),
    F { a: i32, b: String },
}
```
generates (output formatted)
```ts
type SimpleEnum =
  | { A: string }
  | { B: number }
  | "C"
  | { D: [string, number] }
  | { E: Foo }
  | { F: { a: number, b: string } };
```

Which does seem to match serdes default "externally tagged" representation:
```json
{"A":"foo"}
{"B":5}
"C"
{"D":["bar",123]}
{"E":{"bar":{"field":44}}}
{"F":{"a":42,"b":"baz"}}
```

**Comment by NyxCode** [comment]
though #[ts(inline)] is not behaving like it should

**Comment by NyxCode** [comment]
fixed in v6.0.0
